src/hub/Hub.sol

```diff
@@ -122,8 +124,13 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler, IHubGuar
             hubHelpers.notifyDeposit(poolId, scId, assetId, investor, maxClaims);
 
         if (totalPaymentAssetAmount > 0 || cancelledAssetAmount > 0) {
-            sender.sendFulfilledDepositRequest(
-                poolId, scId, assetId, investor, totalPaymentAssetAmount, totalPayoutShareAmount, cancelledAssetAmount
+            sender.sendRequestCallback(
+                poolId,
+                scId,
+                assetId,
+                RequestCallbackMessageLib.FulfilledDepositRequest(
+                    investor, totalPaymentAssetAmount, totalPayoutShareAmount, cancelledAssetAmount
+                ).serialize()
```

Comment:
From an organization PoV feels weird that we create this submessage, but later the core knows about that. Then, why not leave the message in `MessageLib` as it was?

I think we do not need to move the messages as submessages. Just leave in the spoke side to be able to configure custom handlers for them. But the messages still belong to the core.

Reply:
In terms of them being core: I think you could argue we could build an implementation of Spoke on a new chain (e.g. Solana) without building vaults, and in that case we wouldn't need to implement the request and request callback messages. So it is decoupled from the core in that sense.